### PR TITLE
adding user info form

### PR DESCRIPTION
**Pull request by Melkyas**
I have created are /app/additionalinfo.tsx and /components/forms/postRegistration.tsx. I have made some modifications in /app/(auth)/home.tsx, /app/(auth)/register.tsx, /components/auth/forms/registrationForm.tsx and /styles/global.ts.

please leave a comment and I will work on it the next two days or before our next meeting or fell free to make any change .

### CODE REVIEW

File: gui/app/(auth)/register.tsx [hunk 1]
```diff
@@ -25,4 +32,11 @@ const styles = StyleSheet.create({
         alignItems: 'center',
         justifyContent: 'center',
     },
+    body:{
+        width:'100%',
+        height:'100%',
+        backgroundColor:'rgba(52, 52, 52, 0.8)',
```
Comment (elliottallmann) [inline gui/app/(auth)/register.tsx:38]: If we're having multiple themes, is this color something that looks visible on all of them?

File: gui/components/auth/forms/registrationForm.tsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+import { useState } from "react";
+import { useRouter } from "expo-router";
 
 
 
 const registrationSchema = z.object({
     name: z.string().min(2, { message: "Name must be at least 2 characters" }),
     email: z.string().email({ message: "Invalid email address" }),
-    password: z.string().min(16, { message: "Password must be at least 16 characters" }),
+    password: z.string().min(8, { message: "Password must be at least 16 characters" }),
```
Comment (elliottallmann) [inline gui/components/auth/forms/registrationForm.tsx:16]: Password minimum is 16 right now.

Comment (rushneighbors) [reply gui/components/auth/forms/registrationForm.tsx:16]: If changing to 8 characters, the text will need to be changed in the message as well and I'll need to change it in all of the BDD registration tests.

Comment (elliottallmann) [reply gui/components/auth/forms/registrationForm.tsx:16]: Yes, there should be an edit to the tests directory as well.

File: gui/components/forms/postRegistration.tsx
Comment (rushneighbors) [inline gui/components/forms/postRegistration.tsx:1]: Added "Other" to Sex; Changed "Sex" to "Gender"; Changed "Orientation" to a picking picking gender(s) you're interested in category. See attached code (change file type from txt to tsx, it won't let me attach a tsx file).
[postRegistration.txt](https://github.com/KSU-SWE6733-Spring2023-Team3/development-project/files/11206399/postRegistration.txt)

File: gui/components/forms/postRegistration.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  const [interests, setInterests] = useState(null)
+  const [interestsDegree, setInterestsDegree] = useState({})
+  const [zip, setZip] = useState('')
+  const [orientation, setOrientation] = useState('')
+  const [age, setAge] = useState('')
+
+  const handleDegree = (interest, degree) => {
+    const arr = [...interestsDegree]
+    const index = interestsDegree.findIndex(obj => obj.interest === interest);
+    console.log(index)
```
Comment (elliottallmann) [inline gui/components/forms/postRegistration.tsx:45]: Please remove the console log.

File: gui/components/forms/postRegistration.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+  const handleSubmit = () => {
+    const data = {
+      interests: interestsDegree,
+      sex,
+      age,
+      zip,
+      orientation
+    }
+    console.log(data)
```
Comment (elliottallmann) [inline gui/components/forms/postRegistration.tsx:62]: Please remove the console log.

File: gui/context/auth.tsx [hunk 5]
```diff
@@ -13,11 +13,11 @@ function useProtectedRoute(user) {
...
-        if (!user && !inAuthGroup) {
-            router.replace('/home');
-        } else if (user && inAuthGroup) {
-            router.replace('/');
-        }
+        // if (!user && !inAuthGroup) {
+        //     router.replace('/home');
+        // } else if (user && inAuthGroup) {
+        //     router.replace('/');
+        // }
```
Comment (elliottallmann) [inline gui/context/auth.tsx:20]: Why are we removing this?

File: gui/components/forms/postRegistration.tsx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+  { value: 5, label: 'More than once a week' },
+]
+
+const experienceInit = [
+  {value:1, label:'Novice'},
+  {value:2, label:'Beginner'},
+  {value:3, label:'Intermediate'},
+  {value:4, label:'Expert'},
+  {value:5, label:'Master'}
+]
```
Comment (elliottallmann) [inline gui/components/forms/postRegistration.tsx:28]: Please get these values from the API to build the dropdown values. 

Activities: api/activity
Attitudes: api/attitudes
Skill levels: api/skillLevel

These are the values that should go into the drop-downs.

Then, POST a message to api/user/interest that looks like: 
```
[ 
  'interests': [
    { 
      'activity': <activity value>
      'attitude': <attitude value>
      'skillLevel': <skill level value>
    }
    ...
]
```

File: gui/components/forms/postRegistration.tsx
Comment (elliottallmann) [reply gui/components/forms/postRegistration.tsx:1]: Is the intent that postRegistration be what is changed in the repo, or are you just adding it here for reference?

**Pull request merged**